craft_parts/sources/tar_source.py

```diff
@@ -0,0 +1,137 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2015-2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Implement the tar source handler."""
+
+import os
+import re
+import shutil
+import tarfile
+import tempfile
+from typing import Optional
+
+from craft_parts.dirs import ProjectDirs
+
+from . import errors
+from .base import FileSourceHandler
+
+
+# pylint: disable=too-many-arguments
+class TarSource(FileSourceHandler):
+    """The tar source handler."""
+
+    def __init__(
+        self,
+        source,
+        part_src_dir,
+        *,
+        application_name: str = None,
```

Comment:
most of these are Optional[str] instead of str

Reply:
Declared as optional.